python/dgl/graphbolt/graph_storage/sampled_sub_graph.py

```diff
@@ -0,0 +1,156 @@
+"""Heterogeneous sampled subgraph."""
+# pylint: disable= invalid-name
+from collections.abc import Mapping
+
+
+class SampledSubGraph:
+    r"""Class for sampled subgraph."""
+
+    def __init__(
+        self,
+        node_pairs,
+        reverse_column_node_ids=None,
+        reverse_row_node_ids=None,
+        reverse_edge_ids=None,
+    ):
+        """ ""Initialize the SampledSubGraph object. In the context of a
+        heterogeneous graph, each field should be of `Dict` type. For
+        homogeneous graphs, each field should correspond to its respective
+        value type."
+
+        Parameters
+        ----------
+        node_pairs :  Tuple[Tensor] or Dict[(str, str, str), Tuple[Tensor]]
+            Node pairs representing source-destination edges.
+            - If `node_pairs` is a tuple: It should be in the format ('u', 'v')
+                representing source and destination pairs.
+            - If `node_pairs` is a dictionary: The keys should be edge type and
+                the values should be corresponding node pairs. The ids inside
+                is heterogeneous ids.
+        reverse_column_node_ids : Optional[Tensor or Dict[str, Tensor]]
+            Column's reverse node ids in the original graph. A graph structure
+            can be treated as a coordinated row and column pair, and this is
+            the mapped ids of the column.
+            - If `reverse_column_node_ids` is a tensor: It represents the
+                original node ids.
+            - If `reverse_column_node_ids` is a dictionary: The keys should be
+                node type and the values should be corresponding original
+                heterogeneous node ids.
+        reverse_row_node_ids : Optional[Tensor or Dict[str, Tensor]]
+            Row's reverse node ids in the original graph. A graph structure
+            can be treated as a coordinated row and column pair, and this is
+            the mapped ids of the row.
+            - If `reverse_row_node_ids` is a tensor: It represents the
+                original node ids.
+            - If `reverse_row_node_ids` is a dictionary: The keys should be
+                node type and the values should be corresponding original
+                heterogeneous node ids.
+        reverse_edge_ids: Optional[Tensor or Dict[(str, str, str), Tensor]]
+            Reverse edge ids in the original graph.This is useful when edge
+            features are needed.
```

Comment:
Clarify when it will be present and when it will be absent.

Reply:
fixed